src/main/kotlin/com/openlattice/hazelcast/serializers/BulkLinkedDataLambdasStreamSerializer.kt

```diff
@@ -65,25 +60,31 @@ class BulkLinkedDataLambdasStreamSerializer(
         val entityTypeId = readUUID(input)
 
         val linkingIdsSize = input.readInt()
-        val entitiesByLinkingId = HashMap<UUID, Map<UUID, Map<UUID, Set<Any>>>>(linkingIdsSize)
+        val entitiesByLinkingId = HashMap<UUID, Map<UUID, Map<UUID, Map<UUID, Set<Any>>>>>(linkingIdsSize)
         for (i in 1..linkingIdsSize) {
             val linkingId = readUUID(input)
 
             val entitySetsSize = input.readInt()
-            val entitiesByEntitySetId = HashMap<UUID, SetMultimap<UUID, Any>>(entitySetsSize)
+            val entitiesByEntitySetId = HashMap<UUID, Map<UUID, Map<UUID, Set<Any>>>>(entitySetsSize)
             for (j in 1..entitySetsSize) {
                 val entitySetId = readUUID(input)
 
-                val numBytes = input.readInt()
-                var entityData: HashMultimap<UUID, Any>
-                try {
-                    entityData = mapper.readValue(input.readBytes(numBytes), ref)
-                    entitiesByEntitySetId[entitySetId] = entityData
-                } catch (e: IOException) {
-                    logger.debug("Unable to deserialize entities for linking id: {}", linkingId)
+                val originIdsSize = input.readInt()
```

Comment:
I don't see any equivalent updates to the `write()` method for this

Reply:
Ah, nice catch!